src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -89,6 +98,22 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, MSBuild.TaskLog
                     catch (InvalidOperationException ex) when (ex.Message.Contains("init"))
                     {
                         Log.LogWarning($"Sub-feed {source.FeedSubPath} has not been initialized. Initializing now...");
+                        requiresInit = true;
+                    }
+                    catch (InvalidOperationException ex) when (ex.Message.Contains("Unable to obtain a lock on the feed."))
```

Comment:
Why are you checking for string messages? That is troublesome given they might change at anytime and they also might change if they are localized.

Reply:
While I agree there is no other way to catch this particular scenario. This is what being thrown "throw new InvalidOperationException("Unable to obtain a lock on the feed.");" and InvalidOperationException is thrown almost everywhere else